### PR TITLE
`MostViewedRight` as an Island

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
Moves the `MostViewedRight` component away from the `Portal` pattern and uses the `Island` one instead

## Why?
This is part of a wider piece of work to [migrate away](https://docs.google.com/document/d/1kSrZZrJ8hSw6bek5hGa9onE8QDz5TRHFqHV52EOFjts/edit#heading=h.gsod33a0pc47) from `App.tsx`

## What happened to `MostViewRightIsland`?
I also took this opportunity to refactor away the `MostViewRightIsland` component. I think this was needed previously as part of how we hydrated things in two places but now that we're only calling the component in one place we can merged the wrapper css into the main `MostViewedWrapper` component. Note. The reason for the wrapper, and indeed the reason for hydration in the first place, is so that this component can read the height of it's container and use that to decide if it should appear or not.

## Why do we have a flatter file structure?
To make the dynamic import for for the island I had to move these components out of their subfolder and down into the root of `/components`

### CODE REVIEW

File: dotcom-rendering/src/web/components/MostViewedRightWrapper.importable.tsx [hunk 1]
```diff
@@ -11,8 +11,14 @@ type Props = {
...
-const flexGrow = css`
-	flex-grow: 1;
+const MOSTVIEWED_STICKY_HEIGHT = 1059;
+
+// Styling the data island root so it stretches to cover the full height available in the container.
+// Requires us to subtract the height of its sibling in the container (StickyAd).
+const stretchWrapperHeight = css`
+	display: flex;
+	flex-direction: column;
+	height: ${`calc(100% - ${MOSTVIEWED_STICKY_HEIGHT}px)`};
```
Comment (oliverlloyd) [inline dotcom-rendering/src/web/components/MostViewedRightWrapper.importable.tsx:21]: This css was previously being added by `MostViewedRightIsland` but is now included here

File: dotcom-rendering/src/web/components/MostViewedRightWrapper.importable.tsx [hunk 2]
```diff
@@ -11,8 +11,12 @@ type Props = {
...
 const HEIGHT_REQUIRED = 482 + 24 + 24;
 
-const flexGrow = css`
-	flex-grow: 1;
+const MOSTVIEWED_STICKY_HEIGHT = 1059;
+
+// Styling the data island root so it stretches to cover the full height available in the container.
+// Requires us to subtract the height of its sibling in the container (StickyAd).
+const stretchWrapperHeight = css`
+	height: ${`calc(100% - ${MOSTVIEWED_STICKY_HEIGHT}px)`};
```
Comment (oliverlloyd) [inline dotcom-rendering/src/web/components/MostViewedRightWrapper.importable.tsx:19]: We previously achieved this in `MostViewedRightIsland` but I've now merged that file into this one given we only set this component in one place now. This also negates the need for the `flex` styles.

I wonder if those flex styles were ever needed 🤔

File: dotcom-rendering/src/web/components/MostViewedRight.test.tsx [hunk 3]
```diff
@@ -1,14 +1,14 @@
 import { render } from '@testing-library/react';
 
-import { useApi as useApi_ } from '@root/src/web/lib/useApi';
+import { useApi as useApi_ } from '../lib/useApi';
```
Comment (OllysCoding) [inline dotcom-rendering/src/web/components/MostViewedRight.test.tsx:3]: Is there a reason to switch from absolute to relative imports?

**Review by OllysCoding — APPROVED** [review]
+1  👏 Kill 👏 Off 👏 Loadable

File: dotcom-rendering/src/web/components/MostViewedRight.test.tsx (hunk 3 above)
Comment (oliverlloyd) [reply dotcom-rendering/src/web/components/MostViewedRight.test.tsx:3]: There are various benefits to relative paths as well as cons. It's a whole trade offs thing. But I think in most cases it's just easier to allow VSCode to automatically add them. I'm pretty sure this is the reason everyone keeps doing this

**Comment by jamesgorrie** [comment]
part of https://github.com/guardian/dotcom-rendering/pull/3629

**Pull request merged**